app/src/main/java/com/example/iphonesearchapi/model/ResultOf.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.iphonesearchapi.model
+
+
+sealed class ResultOf<out T>() {
+    data class Loading<out R>(val value :R): ResultOf<R>()
+    data class Failure(
+        val message: String?,
+        val throwable: Throwable?
+    ):ResultOf<Nothing>()
+    object Success : ResultOf<Nothing>()
+}
```

Comment:
I think Success and Loading can be switched. As I understand we would have the actual data in the Success state.